boskos/metrics/metrics.go

```diff
@@ -30,58 +32,46 @@ import (
 )
 
 type prometheusMetrics struct {
-	GceStats map[string]prometheus.Gauge
-	GkeStats map[string]prometheus.Gauge
+	BoskosState map[string]map[string]prometheus.Gauge
 }
 
 var (
 	promMetrics = prometheusMetrics{
-		GceStats: map[string]prometheus.Gauge{
-			common.Free: prometheus.NewGauge(prometheus.GaugeOpts{
-				Name: "boskos_gce_project_free",
-				Help: "Number of free gce-project",
-			}),
-			common.Busy: prometheus.NewGauge(prometheus.GaugeOpts{
-				Name: "boskos_gce_project_busy",
-				Help: "Number of busy gce-project",
-			}),
-			common.Dirty: prometheus.NewGauge(prometheus.GaugeOpts{
-				Name: "boskos_gce_project_dirty",
-				Help: "Number of dirty gce-project",
-			}),
-			common.Cleaning: prometheus.NewGauge(prometheus.GaugeOpts{
-				Name: "boskos_gce_project_cleaning",
-				Help: "Number of cleaning gce-project",
-			}),
-		},
-		GkeStats: map[string]prometheus.Gauge{
+		BoskosState: map[string]map[string]prometheus.Gauge{},
+	}
+	resources common.ResTypes
+)
+
+func init() {
+	flag.Var(&resources, "resource-type", "comma-separated list of resources need to have metrics collected")
+}
+
+func init_prometheus() {
```

Comment:
nit: this is not a very go like signature

Reply:
sure